src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/dwpuploadresponse/DwpUploadResponseMidEventValidationHandler.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.dwpuploadresponse;
+
+import static uk.gov.hmcts.reform.sscs.ccd.presubmit.dwpuploadresponse.ElementsDisputed.*;
+
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.EventType;
+import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+
+@Component
+@Slf4j
+public class DwpUploadResponseMidEventValidationHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+
+    PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        return callbackType == CallbackType.MID_EVENT
+            && callback.getEvent() == EventType.DWP_UPLOAD_RESPONSE
+            && Objects.nonNull(callback.getCaseDetails())
+            && Objects.nonNull(callback.getCaseDetails().getCaseData());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        SscsCaseData sscsCaseData = callback.getCaseDetails().getCaseData();
+
+        preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(sscsCaseData);
+
+        checkForDuplicateIssueCodes(sscsCaseData);
+
+        return preSubmitCallbackResponse;
+    }
+
+    private void checkForDuplicateIssueCodes(SscsCaseData sscsCaseData) {
+        String duplicateMessage = " element contains duplicate issue codes";
+
+        if (sscsCaseData.getElementsDisputedGeneral() != null
+                && sscsCaseData.getElementsDisputedGeneral().size() != sscsCaseData.getElementsDisputedGeneral().stream().distinct().count()) {
+            preSubmitCallbackResponse.addError(GENERAL.getValue() + duplicateMessage);
+        }
+        if (sscsCaseData.getElementsDisputedSanctions() != null
+                && sscsCaseData.getElementsDisputedSanctions().size() != sscsCaseData.getElementsDisputedSanctions().stream().distinct().count()) {
```

Comment:
Is this logic intended to check for duplicate issue codes, or distinct "issue code/outcome" pairs?   At the moment it would prevent duplicate pairs,  but allow for two entries with the same issue code but different outcomes.  See comment for the test below.

Reply:
We're not using outcome yet - i've just added to the common to future proof it. It won't be part of this event anyway. Each element can have up to 3 issue codes and these must be unique, within each element.

I think I see what you mean, I will fix.